docs/docs/glossary.md

```diff
@@ -2,7 +2,240 @@
+
+A [query](#query) language that allows you to pull data into your website or app. It’s the interface that Gatsby uses for managing site data.
+
+## H
+
+### Hosting
+
+A hosting provider keeps a copy of your website or app and makes it accessible to [the public](#public).
+
+### HTML
+
+A markup language that every web browser is able to understand. It stands for HyperText Markup Language. HTML gives your web content a universal informational structure, defining things like headings, paragraphs, and more.
+
+### Headless CMS
+
+A [CMS](#cms) that only handles the [back-end](#back-end) content management instead of handling both the back-end and [front-end](#frontend).
+
+## I
+
+## J
+
+### JavaScript
+
+A programming language that helps us make the web dynamic and interactive.
+
+## K
+
+## L
+
+## M
+
+### Markdown
+
+A way of writing content in plain text, using special characters such as underscores and asterisks to format content.
+
+### MDX
+
+Extends [Markdown](#markdown) to support [React](#react) [components](#components) within your content.
+
+## N
+
+### Node
+
+This might refer to Node.js - a program that lets you run [JavaScript](#javascript) on your computer. Gatsby is powered by Node.
+
+In Gatsby it may also refer to a data node. A single piece of data. A [data source](#data-source) will create multiple nodes.
+
+### NPM
+
+[Node](#node) Package Manager. Allows you to install and update other packages that your project depends on. [Gatsby](#gatsby) and [React](#react) are examples of your project's dependencies. See also: [Yarn](#yarn).
+
+## O
+
+## P
+
+### Page
+
+An [HTML](#html) page.
+
+This also often refers to [components](#components) that live in `/src/pages/` and are converted to pages by Gatsby.
+
+### Plugin
+
+Additional code that adds functionality to Gatsby that wasn't included out-of-the-box.
+
+### Public
+
+This usually refers to either a member of the public (as opposed to your team) or the folder `/public` in which your [built](#build) website or app is saved.
+
+### Programmatically
+
+Something that automatically happens based on your code and configuration. For example, you might [configure](#config) your project to create a [page](#page) for every blog post written.
+
+### Production Environment
+
+The [environment](#environment) for the [built](#build) website or app. It's accessed through the [CLI](#cli) using `gatsby build` or `gatsby serve`.
+
+## Q
+
+### Query
+
+The process of requesting specific data from somewhere. With Gatsby you normally query with [GraphQL](#graphql).
+
+## R
+
+### React
+
+A code library (built with [JavaScript](#javascript)) for building user interfaces. It’s the framework that [Gatsby](#gatsby) uses to build pages and structure content.
+
+## S
+
+### Source Code
+
+Source code is your code that lives in `/src/` folder and makes up the unique aspects of your website or app. It is made up of [JavaScript](#javascript) and sometimes [CSS](#css) and other files.
```

Comment:
Would this include Markdown files? Or should we add a Content section for those?

Reply:
I've note really used Markdown on Gatsby so I'm unsure on where the conventional place is to keep them. What would you recommend?

Reply:
On second thought, I think it's fine as-is. 👍